src/server.py

```diff
@@ -1,8 +1,214 @@
+                    "day": meeting["days"],
+                    "date": meeting_date_convert(meeting["dates"]),
+                    "time": {
+                        "start": meeting_time_convert(meeting["start_time"]),
+                        "end": meeting_time_convert(meeting["end_time"])
+                    },
+                    "location": meeting["location"]
+                })
+            
+            cur_class["meetings"] = meetings
+            classes.append(cur_class)
+            
+        cur_class_details["classes"] = classes
+        classes_formatted.append(cur_class_details)
+    
+    course_info = {
+        "id": id,
+        "name": {
+            "subject": details["SUBJECT"],
+            "code": details["CATALOG_NBR"],
+            "title": details["COURSE_TITLE"]
+            # Below removed because such info is not in the courseplanner api
+            #"other_names": ["ADDS", "DSA", "Data structures and algorithms"], # case insensitive in frontend search
+        },
+        "class_number": details["CLASS_NBR"],
+        "year": details["YEAR"],
+        "term": details["TERM_DESCR"], # Could use the "TERM" field but would then need to match the given term code
+        "campus": details["CAMPUS"],
+        "career": details["ACAD_CAREER_DESCR"], # Using a full text version instead of a code (eg UGRD)
+        "units": details["UNITS"],
+        "requirement": {
+            "restriction": details["RESTRICTION_TXT"] if details["RESTRICTION"] == "Y" else None,
+            # Requisites are parsed into a dict with the original string and parsed subjects
+            "prerequisite": parse_requisites(details["PRE_REQUISITE"]),
+            "corequisite": parse_requisites(details["CO_REQUISITE"]),
+            "assumed_knowledge": parse_requisites(details["ASSUMED_KNOWLEDGE"]),
+            "incompatible": parse_requisites(details["INCOMPATIBLE"])
+        },
+        # Not in the spec
+        "description": details["SYLLABUS"],
+        "assessment": details["ASSESSMENT"],
+        "contact": details["CONTACT"], # Could parse to a float of hours, takes the format of "Up to [hours] hours per week"
+        "critical_dates": { # These come in the format of "[3-char weekday] DD/MM/YYYY"
+            "last_day_add_online": details["CRITICAL_DATES"]["LAST_DAY"],
+            "census_date": details["CRITICAL_DATES"]["CENSUS_DT"],
+            "last_day_wnf": details["CRITICAL_DATES"]["LAST_DAY_TO_WFN"],
+            "last_day_wf": details["CRITICAL_DATES"]["LAST_DAY_TO_WF"]
+        },
+        "outline_url": details["URL"],
+        "class_list": classes_formatted
+    }
+    
+    return course_info
+
+def get_courses(year: int, term: str) -> list[dict]:
+    # Make sure year is within bounds
+    if year < 2006 or year > current_year():
+        raise SyntaxError(f"Invalid year: {year}")
+    
+    MAX_RESULTS = 5000
+    term_url = "&term=" + get_term_number(year, term) if term != None else ""
```

Comment:
Generally, use `is None` over `== None` and `is not None` over `!= None`.

More reading: https://stackoverflow.com/questions/3257919/what-is-the-difference-between-is-none-and-none